### PR TITLE
plugins: Surface decision log errors via status API

### PR DESCRIPTION
**Pull request by ashutosh-narkar**
Currently errors encountered during decision log uploads are logged. This change now includes these errors as part of status updates  which can be useful for the control plane to determine if OPA is having any issues while processing, uploading logs etc.

Fixes: #5637

### CODE REVIEW

File: plugins/logs/status/status.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+const (
+	errCode = "decision_log_error"
+)
+
+// Status represents the status of processing a decision log.
+type Status struct {
+	Code     string          `json:"code,omitempty"`
+	Message  string          `json:"message,omitempty"`
+	HTTPCode json.Number     `json:"http_code,omitempty"`
```
Comment (srenatus) [inline plugins/logs/status/status.go:25]: A plain int would do, too, wouldn't it?

File: plugins/status/plugin.go [hunk 2]
```diff
@@ -31,32 +32,35 @@ type Logger interface {
...
-	Bundle    *bundle.Status             `json:"bundle,omitempty"` // Deprecated: Use bulk `bundles` status updates instead
-	Bundles   map[string]*bundle.Status  `json:"bundles,omitempty"`
-	Discovery *bundle.Status             `json:"discovery,omitempty"`
-	Metrics   map[string]interface{}     `json:"metrics,omitempty"`
-	Plugins   map[string]*plugins.Status `json:"plugins,omitempty"`
+	Labels       map[string]string          `json:"labels"`
+	Bundle       *bundle.Status             `json:"bundle,omitempty"` // Deprecated: Use bulk `bundles` status updates instead
+	Bundles      map[string]*bundle.Status  `json:"bundles,omitempty"`
+	Discovery    *bundle.Status             `json:"discovery,omitempty"`
+	DecisionLogs *lstat.Status              `json:"decision_logs,omitempty"`
```
Comment (srenatus) [inline plugins/status/plugin.go:39]: The omitempty here also covers our bases for logging that doesn't return errors, like console output and DL plugins, I suppose?

**Review by srenatus — APPROVED** [review]

File: plugins/logs/status/status.go (hunk 1 above)
Comment (ashutosh-narkar) [reply plugins/logs/status/status.go:25]: It should but just wanted to use what we have [here](https://sourcegraph.com/github.com/open-policy-agent/opa@8aa60e8b525f060ded2c31bba15f11bf82f7eb7a/-/blob/plugins/bundle/status.go?L37).

File: plugins/logs/plugin.go [hunk 3]
```diff
@@ -767,6 +770,19 @@ func (p *Plugin) loop() {
 func (p *Plugin) doOneShot(ctx context.Context) error {
 	uploaded, err := p.oneShot(ctx)
 
+	p.mtx.Lock()
+	defer p.mtx.Unlock()
+
```
Comment (ashutosh-narkar) [inline plugins/logs/plugin.go:775]: Added this change to address the [race detector](https://github.com/open-policy-agent/opa/actions/runs/4351330130/jobs/7619374942#step:4:222) failure.

File: plugins/logs/status/status.go (hunk 1 above)
Comment (srenatus) [reply plugins/logs/status/status.go:25]: Ok!

**Review by srenatus — APPROVED** [review]

**Pull request merged**